src/logcollector/logcollector.c

```diff
@@ -1801,10 +1801,30 @@ void * w_output_thread(void * args){
         message = w_msg_queue_pop(msg_queue);
 
         if (SendMSGtoSCK(logr_queue, message->buffer, message->file, message->queue_mq, message->log_target) < 0) {
-            merror(QUEUE_SEND);
+            if (strcmp(message->log_target->log_socket->name, "agent") == 0) {
+                // When dealing with this type of messages we don't want any of them to be lost
+                // Continuously attempt to reconnect to the queue and send the message.
+                int sleep_time = 5;
+
+                while(1) {
+                    if(logr_queue = StartMQ(DEFAULTQPATH, WRITE), logr_queue > 0) {
+                        if (SendMSGtoSCK(logr_queue, message->buffer, message->file, message->queue_mq, message->log_target) == 0) {
+                            break;  //  We sent the message successfully, we can go on.
```

Comment:
When the daemon is reconnected and able to send events to the queue again, it would be interesting to log it.

Reply:
Done